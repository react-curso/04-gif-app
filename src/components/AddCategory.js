import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ categories }) => { 

     const [inputValue, setInputValue] = useState('');

     const handleInputChange = (e) => {
          setInputValue(e.target.value);
     }

     const handleSubmit = (e) => {
          e.preventDefault();
          if (inputValue.trim().length > 2) {
               categories(c => [ inputValue, ...c]);
               setInputValue('');

               
          }
     }

     return (
          <>
               <form onSubmit={handleSubmit} >
                    <input
                         type="text"
                         value={inputValue}
                         onChange={handleInputChange}
                    />
               </form>
          </>
     )
}

AddCategory.propTypes = {
     categories: PropTypes.func.isRequired
}

