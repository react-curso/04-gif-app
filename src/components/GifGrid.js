import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            { loading && <p>Loading</p> }            

            <div className="card-grid">
                {
                    images.map((img) => (

                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}
