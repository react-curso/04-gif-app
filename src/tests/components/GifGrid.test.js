import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import '@testing-library/jest-dom'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test the component GifGrid', () => {
    
    const category = 'Goku';
    
    test('test rederer succeful', () => {

        useFetchGifs.mockReturnValue({
            images:[],
            loading:true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar item al cargar imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: 'abc',
            url: 'https://gid.png',
            title: 'hola mundo'
        }];
        
        useFetchGifs.mockReturnValue({
            images: gifs,
            loading:false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


    })
    
    

})
