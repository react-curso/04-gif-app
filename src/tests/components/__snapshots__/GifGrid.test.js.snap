// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the component GifGrid debe de mostrar item al cargar imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="hola mundo"
      url="https://gid.png"
    />
  </div>
</Fragment>
`;

exports[`test the component GifGrid test rederer succeful 1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
